File: python/ticket.py
# This represents a single ticket

class Ticket:
    def __init__(self, id, passenger_id, passenger_name, flight_id, origin_city, 
    destination_city, departure_time, departure_gate, checkin_time, carousel):
        self.id = id
        self.passenger_id = passenger_id
        self.passenger_name = passenger_name
        self.flight_id = flight_id
        self.origin_city = origin_city
        self.destination_city = destination_city
        self.departure_time = departure_time
        self.departure_gate = departure_gate
        self.checkin_time = checkin_time
        self.carousel = carousel

    def print(self):
        print("ticket: " + self.id + " for flight " + self.flight_id + " - From: " + self.origin_city + "  To: " + self.destination_city + 
            " departing at " + self.departure_time + " from gate " + self.departure_gate)

    # Return this record as a single line in a CSV file
    def as_csv(self):
        return f"{self.id},{self.passenger_id},{self.passenger_name},{self.flight_id},{self.origin_city}," + \
            f"{self.destination_city},{self.departure_time},{self.departure_gate},{self.checkin_time},{self.carousel}\n"
        
    # Return this record as a single line in a CSV file
    @staticmethod
    def csv_header():
        return "id,passenger_id,passenger_name,flight_id,origin_city," + \
            "destination_city,departure_time,departure_gate,checkin_time,carousel\n"

# create table ticket(
#     id uuid,
#     passenger_id uuid,
#     passenger_name text,
#     flight_id text,
#     origin_city text,
#     destination_city text,
#     departure_time timestamp,
#     departure_gate text,
#     checkin_time text,
#     carousel text,
#     PRIMARY KEY(id)
# );

File: flutter_app/lib/entities/Ticket.dart
class Ticket {
  late String id;
  late String departure_gate;
  late String departure_time;
  late String destination_city;
  late String origin_city;
  late String checkin_time;
  late String flight_id;
  late String passenger_id;
  late String passenger_name;
  late int carousel;

  Ticket() {}

  Ticket.fromJson(Map<String, dynamic> json) {
    id = json['id'] as String;
    departure_gate = json['departure_gate'] as String;
    departure_time = json['departure_time'] as String;
    destination_city = json['destination_city'] as String;
    origin_city = json['origin_city'] as String;
    checkin_time = json['checkin_time'] as String;
    flight_id = json['flight_id'] as String;
    passenger_id = json['passenger_id'] as String;
    passenger_name = json['passenger_name'] as String;
    carousel = json['carousel'] as int;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['departure_gate'] = departure_gate;
    data['departure_time'] = departure_time;
    data['destination_city'] = destination_city;
    data['origin_city'] = origin_city;
    data['checkin_time'] = checkin_time;
    data['flight_id'] = flight_id;
    data['carousel'] = carousel;
    return data;
  }
}
